### ISSUE TITLE
Adding Unwanted Shows to Trakt Collection

### ISSUE THREAD

**Issue by Wanderer189** [razzeee/script.trakt#461]
I been trying to figure this out for months. For some reason when I run trakt manually from within Kodi, it would add strange shows to my online trakt collection. I think with the help of the folks at Trakt, we have figured it out. Here is an example of what is happening:

I add a show to my local library, gonna use the show Happy! TMDB ID=72339. All works fine, Kodi Scrapes it correct, etc. So now I want to run trakt so it will mark watched episodes, etc. That works, but here is what happens, it then adds a new show to my online trakt collection called "15 Stories High". I can delete the new show from my trakt collection, run trakt again and it puts it back every time. 

I have 2 Shields, each with different shows and trakt accounts, with about 150 shows between the 2 of them, this bug is resulting in 5 or 6 shows being added to the both of them. The files are stored on a NAS.

At first I thought it was because the special character in the name, in this example the "!", as most of them have a special character, which is being replaced with an underscore, such as Happy_.

But working with the folks at Trakt, they ran across something else. If you look at the show Happy!, it's, TMDB ID=72339, then if you go to the TVDB and look up the problem show (15 Stories High), it is using the same ID number, 72339. So when I run trakt, it finds that number for Happy, and when it tries to add new shows found in my local library to my trakt collection, it add both shows.

If you use a 3rd party addon that allows you to sync your trakt collection to your local library, you will get extra, unwanted shows added.

Any ideas on how to fix this?

**Comment by rectifyer** [comment]
Which scraper are you using in Kodi?

**Comment by Wanderer189** [comment]
Haven't checked, which ever one is the default in Kodi v18.1
Update: Just checked it, TMDB is the one being used

UPDATE #2: I changed the scraper to TVDB and it appears to have stopped it on inital test. Will keep testing.

**Comment by NixSar** [comment]
I can confirm this behavior for Happy! and it happens for other TV shows, for example Tuca & Bertie (TMDB 79356) resolves to Brotherhood (TVDB 79356). This can be resolved by selecting TVDB as the default scraper, but is really unwanted behavior since it arises from all default settings in Kodi 18 and Trakt plugin.

**Comment by Wanderer189** [comment]
Yes, it happens with any TV show that has TVDV and TMDB using the same number as it's ID. It seems to mess with addons that utilize library functions using the TMDB if you switch Kodi to use TVDB for it's scraper, not sure, but seems to have with mine running v18.2

**Comment by NixSar** [comment]
It seems the proper solution for this would be for Kodi database to include both TVDB and TMDB show IDs and let plugins decide which one to use.

**Comment by razzeee** [comment]
@nsarnav that was already done as far as i know. But nobody updated this addon to handle this correctly, yet.
